plugins/defaults.js

```diff
@@ -0,0 +1,45 @@
+const Plugin = require('./base');
+const { Scopes } = require('../lib/constants');
+
+const setDefaults = (options = {}, defaults) => {
+    return Object.keys(options)
+        .reduce((acc, key) => {
+            if (acc[key] !== null &&
+                options[key] !== null &&
+                typeof acc[key] === 'object' &&
+                typeof options[key] === 'object') {
```

Comment:
Maybe you can extract `!= null && typeof === 'object'`in a separate function like `isObject` to improve a bit the readability